include/wsrep/transaction.hpp

```diff
@@ -240,6 +248,7 @@ namespace wsrep
         size_t fragments_certified_for_statement_;
         wsrep::streaming_context streaming_context_;
         wsrep::sr_key_set sr_keys_;
+        wsrep::mutable_buffer apply_error_buf_;
```

Comment:
Why this is part of `transaction` class? More suitable would be `client_state` (TOI operations do not have active transaction for example).

Reply:
1. `client state` does not seem in need to operate directly with provider, and there seems to be no benefit of having it there. `leave_toi()` is one thing, but given that TOI processing takes a very different and much simpler path, having error buffer as a member there doesn't seem to serve any particular purpose.
2. Cleanup. Having it in `transaction` means very clear cleanup approach.

Overall I see that moving it to 'client_state' will only complicate the code without any obvious benefits.